src/main/java/core/basesyntax/converter/DataConverterImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.converter;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : inputReport) {
+            String[] parts = line.split(",");
+            if (parts.length != 3) {
+                throw new IllegalArgumentException("Invalid line format: " + line);
+            }
+            String operationCode = parts[0].trim();
+            String fruit = parts[1].trim();
+            int quantity;
+            try {
+                quantity = Integer.parseInt(parts[2].trim());
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException("Invalid quantity in line: " + line);
+            }
+            FruitTransaction.Operation operation = FruitTransaction.Operation
+                    .getOperation(operationCode);
```

Comment:
Ensure that the `getOperation` method in `FruitTransaction.Operation` is correctly implemented to handle invalid operation codes. It should throw an `IllegalArgumentException` if the operation code does not match any known operation.